File: c/tree/balanced-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

int my_abs(int a, int b)
{
    if(a > b)
        return a - b;
    return b - a;
}
bool getResult(struct TreeNode *root, int *height)
{
    if(root == NULL){
        *height = 0;
        return true;
    }

    int left = 0;
    int right = 0;
    bool bleft = getResult(root->left,&left);
    bool bright = getResult(root->right,&right);
    bool isbalanced = my_abs(left,right) <= 1;

    *height = (left > right ? left : right)+1;
    return bleft && bright && isbalanced;
}

bool isBalanced(struct TreeNode* root) {
    int height;
    return getResult(root,&height);
}

File: python/tree/balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def depth(self,root):
        if not root:
            return 0
        left = self.depth(root.left)
        right = self.depth(root.right)
        if left == -1 or right == -1 or \
            abs(left-right) > 1:
            return -1
        return max(left,right) + 1

    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        return self.depth(root) != -1
